it/v3/endpoints/TriggerBsasControllerISpec.scala

```diff
@@ -19,47 +19,56 @@ package v3.endpoints
 import com.github.tomakehurst.wiremock.stubbing.StubMapping
 import play.api.http.HeaderNames.ACCEPT
 import play.api.http.Status._
-import play.api.libs.json.{JsObject, Json}
-import play.api.libs.ws.{WSRequest, WSResponse}
+import play.api.libs.json.{ JsObject, Json }
+import play.api.libs.ws.{ WSRequest, WSResponse }
 import play.api.test.Helpers.AUTHORIZATION
 import support.IntegrationBaseSpec
 import v3.fixtures.TriggerBsasRequestBodyFixtures._
 import v3.models.errors._
-import v3.stubs.{AuditStub, AuthStub, DownstreamStub, MtdIdLookupStub}
+import v3.stubs.{ AuditStub, AuthStub, DownstreamStub, MtdIdLookupStub }
 
 class TriggerBsasControllerISpec extends IntegrationBaseSpec {
 
   private trait Test {
 
     val nino = "AA123456A"
 
-    def setupStubs(): StubMapping
+    def taxYear: String
     def uri: String = s"/$nino/trigger"
-    def desUrl: String = s"/income-tax/adjustable-summary-calculation/$nino"
+    def downstreamUri: String
+    def downstreamTaxYear: String
+
+    def setupStubs(): StubMapping
 
     def request(): WSRequest = {
       setupStubs()
       buildRequest(uri)
         .withHttpHeaders(
           (ACCEPT, "application/vnd.hmrc.3.0+json"),
           (AUTHORIZATION, "Bearer 123") // some bearer token
-      )
+        )
     }
 
     def errorBody(code: String): String =
       s"""
          |      {
          |        "code": "$code",
-         |        "reason": "des message"
+         |        "reason": "message"
          |      }
     """.stripMargin
 
-
     def requestBody(typeOfBusiness: String): JsObject =
       Json.obj(
         "accountingPeriod" -> Json.obj("startDate" -> "2019-01-01", "endDate" -> "2022-10-31"),
         "typeOfBusiness"   -> typeOfBusiness,
-        "businessId" -> "XAIS12345678901"
+        "businessId"       -> "XAIS12345678901"
+      )
+
+    def tysRequestBody(typeOfBusiness: String): JsObject =
```

Comment:
Could we rename `requestBody` to `makeRequestBody` and then pass in a start year and end year so we only need one function? Seems pointless having two functions

Reply:
Good idea. Instead of start date and end date as parameters, I've just added `tys` as a boolean. Let me know what you think